cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/Schedulers.java

```diff
@@ -147,10 +153,28 @@ public ScheduleDetail apply(@Nullable ProgramSchedule input) {
     });
   }
 
-  public static StreamSizeSchedule toStreamSizeSchedule(ProgramSchedule schedule) {
-    StreamSizeTrigger trigger = (StreamSizeTrigger) schedule.getTrigger();
-    return new StreamSizeSchedule(schedule.getName(), schedule.getDescription(),
-                                  trigger.getStreamId().getStream(), trigger.getTriggerMB());
+  /**
+   * Convert a {@link ScheduleTrigger} to the corresponding {@link SatisfiableTrigger}
+   */
+  public static Trigger toSatisfiableTrigger(ScheduleTrigger trigger) {
```

Comment:
Isn't there a more polymorphic way of doing this rather than checking the type and calling the same method based on the type?